src/commands/infra/generate.test.ts

```diff
@@ -31,206 +142,225 @@ describe("test dirIteration", () => {
-      "custerName": "leaf",
-      "variable1": "leaf1",
-      "variable2": "leaf2",
-      "xextra": "xextra"
+      custerName: "leaf",
+      variable1: "leaf1",
+      variable2: "leaf2",
+      xextra: "xextra"
     });
   });
   it("multiple variables test, leafObject has more values", () => {
     const parentObject = {
-      "custerName": "parent",
-      "variable1": "parent1",
-      "variable2": "parent2"
+      custerName: "parent",
+      variable1: "parent1",
+      variable2: "parent2"
     };
     const leafObject = {
-      "custerName": "leaf",
-      "variable1": "leaf1",
-      "variable2": "leaf2",
-      "xextra": "xextra"
+      custerName: "leaf",
+      variable1: "leaf1",
+      variable2: "leaf2",
+      xextra: "xextra"
     };
     const result = dirIteration(parentObject, leafObject);
     expect(result).toEqual(leafObject);
   });
-})
+});
 
 describe("Validate sources in definition.yaml files", () => {
-  test("Validating that a provided project folder contains definition.yaml files with valid source, version, and template", async () => {
-    let mockParentPath = "src/commands/infra/mocks/discovery-service";
-    let mockProjectPath = "src/commands/infra/mocks/discovery-service/west";
-    const expectedArrayWest = [
-      "A",
-      "https://github.com/yradsmikham/spk-source",
-      "cluster/environments/azure-single-keyvault",
-      "v0.0.2"
-    ];
-    let sourceConfiguration = await validateDefinition(
+  test("definition.yaml of leaf override parent's variable", async () => {
+    const mockParentPath = "src/commands/infra/mocks/discovery-service";
+    const mockProjectPath = "src/commands/infra/mocks/discovery-service/west";
+    const expectedSourceWest = {
+      source: "https://github.com/yradsmikham/spk-source",
+      template: "cluster/environments/azure-single-keyvault",
+      version: "v0.0.2"
+    };
+    const sourceConfiguration = validateDefinition(
       mockParentPath,
       mockProjectPath
     );
-    let returnArray = await validateTemplateSources(
+    expect(sourceConfiguration).toEqual(DefinitionYAMLExistence.BOTH_EXIST);
+    const sourceData = validateTemplateSources(
       sourceConfiguration,
       path.join(mockParentPath, `definition.yaml`),
       path.join(mockProjectPath, `definition.yaml`)
     );
-    expect(returnArray).toEqual(expectedArrayWest);
-    await generateConfig(mockParentPath, mockProjectPath, returnArray);
-
-    mockProjectPath = "src/commands/infra/mocks/discovery-service/east";
-    const expectedArrayEast = [
-      "B",
-      "https://github.com/yradsmikham/spk-source",
-      "cluster/environments/azure-single-keyvault",
-      "v0.0.1"
-    ];
-    sourceConfiguration = await validateDefinition(
+    expect(sourceData).toEqual(expectedSourceWest);
+    await generateConfig(
+      mockParentPath,
+      mockProjectPath,
+      sourceConfiguration,
+      sourceData
+    );
+  });
+  test("without parent's definition.xml", async () => {
```

Comment:
@dennisseah you mean yaml instead of xml?

Reply:
yes. thanks